src/main/java/com/android/volley/toolbox/DiskBasedCacheUtility.java

```diff
@@ -277,6 +286,21 @@ static void writeHeaderList(@Nullable List<Header> headers, ByteBuffer buffer)
         return result;
     }
 
+    static List<Header> readHeaderList(ByteBuffer buffer) throws IOException {
+        int size = buffer.getInt();
+        if (size < 0) {
+            throw new IOException("readHeaderList size=" + size);
+        }
+        List<Header> result =
+                (size == 0) ? Collections.<Header>emptyList() : new ArrayList<Header>();
```

Comment:
I would keep this simple and just do new ArrayList<Header>(size). Optimizing to return emptyList() is actually a sort of anti-pattern here - it means that this method sometimes returns an immutable object (when empty) and sometimes doesn't (when non-empty), which can lead to bugs.